dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/provider/ProviderFactory.java

```diff
@@ -1684,8 +1765,10 @@ protected static boolean filterContractSupported(ProviderInfo<?> provider,
 
     public void setProviderComparator(Comparator<?> providerComparator) {
         this.providerComparator = providerComparator;
-        sortReaders();
-        sortWriters();
+        //Liberty code change start
+        addAndSortReaders(new ArrayList<ProviderInfo<MessageBodyReader<?>>>(0));
+        addAndSortWriters(new ArrayList<ProviderInfo<MessageBodyWriter<?>>>(0));
```

Comment:
I don't like these two lines - instead of creating a new empty list, would it be possible to pass in an empty list constant?  Or maybe just sort the readers and writers directly without adding an empty list?